pkgs/top-level/python-packages.nix

```
@@ -5268,7 +5268,7 @@ in {
   wxPython = self.wxPython30;
 
   wxPython30 = callPackage ../development/python-modules/wxPython/3.0.nix {
-    wxGTK = pkgs.wxGTK30;
+    wxGTK = pkgs.wxGTK30.override { withGtk2 = false; };
```

isn't this exactly what you're trying to avoid?
```suggestion
    wxGTK = pkgs.wxGTK30;
```